plugins/hosts/darwin/cap/fs_iso.rb

```diff
@@ -0,0 +1,62 @@
+require 'pathname'
+require "vagrant/util/subprocess"
+require "vagrant/util/map_command_options"
+require "vagrant/util/directory_changed"
+require "vagrant/util/ensure_output_file"
+
+module VagrantPlugins
+  module HostDarwin
+    module Cap
+      class FsISO
+        extend Vagrant::Util::EnsureOutputFile
+
+        @@logger = Log4r::Logger.new("vagrant::host::darwin::fs_iso")
+
+        BUILD_ISO_CMD = "hdiutil".freeze
+
+        # Check that the host has the ability to generate ISOs
+        #
+        # @param [Vagrant::Environment] env
+        # @return [Boolean]
+        def self.isofs_available(env)
+          !!Vagrant::Util::Which.which("hdiutil")
+        end
+
+        # Generate an ISO file of the given source directory
+        #
+        # @param [Vagrant::Environment] env
+        # @param [String] source_directory Contents of ISO
+        # @param [String, nil] file_destination Location to store ISO
+        # @param [Map] extra arguments to pass to the iso building command
+        # @return [Pathname] ISO location
+        # @note If file_destination exists, source_directory will be checked
+        #       for recent modifications and a new ISO will be generated if requried.
+        def self.create_iso(env, source_directory, file_destination=nil, extra_opts={})
+          file_destination = ensure_output_file(ext=".iso", file_destination=file_destination)
+          source_directory = Pathname.new(source_directory)
+
+          # If the destrination does not exist or there have been changes in the source directory since the last build, then build
+          if !file_destination.exist? || Vagrant::Util::DirectoryChanged.directory_changed?(source_directory, file_destination.mtime)
```

Comment:
Thinking about this a little more, there are two distinct conditions when we need to build the ISO:

1. If the destination file doesn't exist
2. If the destination file exists, and source files have changed -- in this case we need to also delete the destination file so that `hdiutil` doesn't throw an error.

```suggestion
          if !file_destination.exist?
            build_required = true
          elsif file_destination.exist? && Vagrant::Util::DirectoryChanged.directory_changed?(source_directory, file_destination.mtime)
            File.delete(file_destination)
            build_required = true
          end

          if build_required
            @@logger.info("Building ISO from source #{source_directory}")
```

Reply:
oh nice catch! I think this can be expressed more succinctly using the `hdiutil` option `-ov` which will overwrite the destination path if it already exists. I think that will cover both conditions.